### DISCUSSION TITLE
New tab doesn't remember path on first try

### DISCUSSION THREAD

**Discussion by 0x4bs3nt** [ghostty-org/ghostty#2744] [Q&A] [answered]
When working with one tab in ghostty and going to a path, for example /DevProjects/project1, before, when opening a new tab, that tab would remember the path of the tab it was opened from (when using shortcuts). 

This no longer works on the first try, but the workaround is to close the new tab and then once again reopen the tab, then the path is remembered correctly.

Video explains what I am talking about:

https://github.com/user-attachments/assets/14e2caea-81d7-4794-886e-36ed71fc213e

This is currently happening on Arch; 6.11.8-arch1-2.

**Ghostty version info:**
Ghostty 0.1.0-main+77ab6448

Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.16.5
    runtime    : 4.16.5
  - libadwaita : enabled
    build      : 1.6.1
    runtime    : 1.6.1

**Comment by 0x4bs3nt** [comment]
Are we sure this isn't an issue, this has worked before, I haven't changed anything to any of my configs?

**Reply by mitchellh** [reply]
I'm not sure it's an issue yet, so it's a discussion until then. Discussions are how we [triage bugs](https://github.com/ghostty-org/ghostty/blob/main/CONTRIBUTING.md) (amongst other things) in this project. They get promoted to issues when they become directly actionable. In this case, there's still quite a lot of understanding to be done.

**Reply by mitchellh** [reply, marked as answer]
Reopened it in #2745 since I was able to reproduce this.

**Comment by mitchellh** [comment]
The way this feature works is that the shell tells Ghostty the pwd, Ghostty stores that, and then when you open a new surface from an existing one, we see if we know the pwd. 

If this isn't working, its only possible in 3 places:

1. The shell isn't sending us the pwd.
2. Ghostty isn't accepting the pwd.
3. New surface isn't reading the pwd.

We need to figure out which one of those three it is. Number 1 is not a Ghostty bug, but 2 or 3 would be.
